core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -459,9 +460,16 @@ public static List<String> getMetadataObjectLocation(
               // The Hive default schema location is Hive warehouse directory
               String defaultSchemaLocation =
                   getHiveDefaultLocation(ident.namespace().level(0), ident.name());
-              if (defaultSchemaLocation != null && !defaultSchemaLocation.isEmpty()) {
+              if (StringUtils.isNotBlank(defaultSchemaLocation)) {
                 locations.add(defaultSchemaLocation);
               }
+            } else if (catalogObj.provider().equals("hadoop")) {
+              String catalogLocation = catalogObj.properties().get(HiveConstants.LOCATION);
```

Comment:
I'm a little surprised that our `core` module relies on `catalog-common` module, which is not what I want, `catalog-common` as its name, should always be used by catalog-* modules and other modules rely on catalog. But for `core` module, the dependency relation is the opposite, we should fix this.

I saw it is introduced by @FANNG1 , so we really should fix it.

Besides, why a hadoop catalog needs to get a Hive property definition? It's weird to me.

Reply:
> Besides, why a hadoop catalog needs to get a Hive property definition? It's weird to me.

I just want to reuse the string constant `location`, indeed it's not so elegant to use constants in HiveConstants, I will introduce a similar value in the `HadoopConstants`

Reply:
We can have a common constant

Reply:
The name for `catalog-common` is a little confusing, by design, it contains the constants shared by catalogs and connectors and Iceberg REST server, etc.

Reply:
Besides, for catalog/connector, the dependency of `catalog-common` is reasonable, because it is the downstream of the `catalog-common` module. But `core` module is the upstream of the `catalog-common` model, so it is not a good implementation.